src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -105,7 +142,49 @@ private static Task WriteToResponseAsync<T>(T? value, HttpContext httpContext, J
+                (false, {{RequestDelegateCreationLogging.RequestBodyIOExceptionEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.RequestBodyIOExceptionMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.InvalidJsonRequestBodyEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.InvalidJsonRequestBodyExceptionMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.ParameterBindingFailedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.ParameterBindingFailedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.RequiredParameterNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.RequiredParameterNotProvidedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.ImplicitBodyNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.ImplicitBodyNotProvidedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.UnexpectedJsonContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedJsonContentTypeLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.UnexpectedFormContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedFormContentTypeLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.InvalidFormRequestBodyEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.InvalidFormRequestBodyLogMessage, true)}},
+                _ => throw new UnreachableException("Encountered unsupported logging EventId.")
+            };
+            void logOrThrowException(int eventId, string eventName, Exception? exception = null, int statusCode = StatusCodes.Status400BadRequest, params object?[] messageFormatArgs)
+            {
+                if ((options?.ThrowOnBadRequest ?? false) && eventId != {{RequestDelegateCreationLogging.RequestBodyIOExceptionEventId}})
+                {
+                    var exceptionMessage = string.Format(CultureInfo.InvariantCulture, getMessageToFormat(true, eventId), messageFormatArgs);
+                    if (exception != null)
+                    {
+                        throw new BadHttpRequestException(exceptionMessage, statusCode, exception);
+                    }
+                    throw new BadHttpRequestException(exceptionMessage, statusCode);
+                }
+                var loggerFactory = serviceProvider?.GetRequiredService<ILoggerFactory>();
+                var rdgLogger = loggerFactory?.CreateLogger("{{typeof(RequestDelegateGenerator)}}");
+                rdgLogger?.LogDebug(new EventId(eventId, eventName), exception, getMessageToFormat(false, eventId), messageFormatArgs);
```

Comment:
Currently it looks like all the logs are debug, but is that always going to be the case?

Reply:
At the moment, all the logs we currently emit in RDF are at the log-level so this feels sufficient for now. If we do need to change the codegen, we can always update this to use `Log` with a `LogLevel` parameter. I don't think this is necessary at the moment though.